code/game/objects/effects/decals/decal.dm

```diff
@@ -9,3 +9,30 @@
 /obj/effect/decal/fire_act(exposed_temperature, exposed_volume)
 	if(!(resistance_flags & FIRE_PROOF)) //non fire proof decal or being burned by lava
 		qdel(src)
+
+
+/obj/effect/turf_decal
+	var/group = TURF_DECAL_PAINT
+	icon = 'icons/turf/decals.dmi'
+	icon_state = "warningline"
+
+//in case we need some special decals
+/obj/effect/turf_decal/proc/get_decal()
+	return image(icon='icons/turf/decals.dmi',icon_state=icon_state,dir=dir,layer=TURF_LAYER)
+
+/obj/effect/turf_decal/initialize()
+	var/turf/T = loc
+	if(!istype(T)) //you know this will happen somehow

```

Comment:
crash("decal on non turf or turf in nullspace")

ends the proc and leaves a runtime with stacktrace